### ISSUE TITLE
Avoid raising "Malformed .swift-version file" error

### ISSUE THREAD

**Issue by MaxDesiatov** [nicklockwood/SwiftFormat#639]
I've got a somewhat peculiar request, but please bear with me.

Versions stored in the `.swift-version` file don't always match semantic versions. As [`swiftenv`](https://github.com/kylef/swiftenv) also relies on encoding versions in the file, it allows installing nightly snapshots with the `swiftenv install` command. For example, [SwiftWasm](https://swiftwasm.org/) relies on that as it is currently [distributed only in the form of nightly snapshots](https://github.com/swiftwasm/swift/releases). Versions stored in `.swift-version` for SwiftWasm look like `wasm-DEVELOPMENT-SNAPSHOT-2020-05-25-a`.

When formatting such project with SwiftFormat, it complains about the version being invalid:

```
error: Malformed .swift-version file
```

even though my `.swiftformat` explicitly specifies the version as `--swiftversion 5.2`.

I can imagine the other use case could be installing [the official nightly snapshots](https://swift.org/download/#snapshots), which have similar version strings. I would expect other people encode a specific snapshot date in the version in their `.swift-version` when testing a feature which only appeared in a recent Swift snapshot.

Overall, I hope SwiftFormat could at least avoid parsing `.swift-version` file in the presence of a valid `--swiftversion` flag. Ideally, it just shouldn't raise a non-zero exit code if it has to parse that file. Non-zero exit codes make things inconvenient for CI runs and pre-commit hooks.

**Comment by MaxDesiatov** [comment]
Hi @nicklockwood, would you mind tagging a new release to include the fix for this? I'm using `swiftformat` as a pre-commit hook in almost all of my projects, and this error requires forcing all the hooks for every commit where a certain Swift snapshot is used. Using `swiftformat` with the fix would save some time there 🙂

Thanks!

**Comment by nicklockwood** [comment]
@MaxDesiatov fixed in 0.44.14.

**Comment by MaxDesiatov** [comment]
Amazing, thank you!

**Comment by MaxDesiatov** [comment]
@nicklockwood this is reproducible again with SwiftFormat 0.47.2. I'm not able to reopen this issue, would you like me to create a new one?

```
% swiftformat .
Running SwiftFormat...
Reading config file at /Users/maxd/Documents/Tokamak/.swiftformat
error: Unrecognized swift version string 'wasm-5.3-SNAPSHOT-2020-10-29-c
' found in file /Users/maxd/Documents/Tokamak/.swift-version.
```

**Comment by nicklockwood** [comment]
Ah sorry, I'll fix it again 🤦‍♂️

**Comment by nicklockwood** [comment]
@MaxDesiatov fixed in 0.47.3

**Comment by MaxDesiatov** [comment]
Thanks!
